[ADD] Sales: Members Backport

pabr-odoo opened the pull request:
Task card: https://www.odoo.com/odoo/project/3835/tasks/4046848

Backporting the Members doc from 17.0 to 16.0.

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/10158.png)](https://mergebot.odoo.com/odoo/documentation/pull/10158)

pabr-odoo commented:
Hey @StraubCreative tagging you for first review of this backport. Thanks!

StraubCreative commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+From the resulting contact form, open the :guilabel:`Membership` tab, and click :guilabel:`Buy
+Membership`.
+
+.. tip::
+   To successfully buy a membership, both the :guilabel:`Account Receivable` and :guilabel:`Account
+   Payable` lines must first be configured for that contact. These can be found in the contact
+   detail form in the :guilabel:`Accounting` tab under the :guilabel:`Accounting entries` heading.
```
QQ: this section is not present in the `17.0` version of this doc, was that on purpose?

pabr-odoo replied:
Yes, when following the flow for 16, I found it was necessary to have these 2 fields configured before buying a membership to prevent an error.

pabr-odoo commented:
Hey @ksc-odoo got another (smaller) one for ya for final review whenever you get the chance. Thanks so much.

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+Members
+=======
+
+The *Members* application is where all operations related to memberships can be configured and
```
```suggestion
The **Members** application is where all operations related to memberships can be configured and
```
using **BOLD** for app names now, based on most recent RST guideline directives

ksc-odoo replied:
Please implement this suggestion, as it is now a requirement when it comes to formatting application names. Thanks

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+=======
+
+The *Members* application is where all operations related to memberships can be configured and
+managed. The *Members* app integrates with the *Sales* and *Accounting* applications to sell and
```
```suggestion
managed. The **Members** app integrates with the **Sales** and **Accounting** applications to sell and
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+To create a new membership product, navigate to :menuselection:`Members app --> Configuration -->
+Membership Products`, and click :guilabel:`New` to open a blank product record.
+
+Complete the blank form with the necessary information, including the :guilabel:`Membership
+Duration`.
```
I think this needs to be expanded upon, and provide the reader with information about all the fields present on the form. Doing so would provide some much-needed context, and add some value to the overall doc.

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+Activate a membership
+=====================
+
+To activate a membership from the *Contacts* application, navigate to :menuselection:`Contacts app`,
```
```suggestion
To activate a membership from the **Contacts** application, navigate to the :menuselection:`Contacts app`,
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+=====================
+
+To activate a membership from the *Contacts* application, navigate to :menuselection:`Contacts app`,
+and click on a contact to open that specific contact's detail form.
```
```suggestion
and click on a contact to open that specific contact form.
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+To activate a membership from the *Contacts* application, navigate to :menuselection:`Contacts app`,
+and click on a contact to open that specific contact's detail form.
+
+From the resulting contact form, open the :guilabel:`Membership` tab, and click :guilabel:`Buy
```
```suggestion
From the contact form, open the :guilabel:`Membership` tab, and click :guilabel:`Buy
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+.. tip::
+   To successfully buy a membership, both the :guilabel:`Account Receivable` and :guilabel:`Account
+   Payable` lines must first be configured for that contact. These can be found in the contact
+   detail form in the :guilabel:`Accounting` tab under the :guilabel:`Accounting entries` heading.
```
```suggestion
   form's :guilabel:`Accounting` tab, under the :guilabel:`Accounting entries` section.
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+Publish members directory
+=========================
+
+To publish a list of active members on the website, the *Online Members Directory* application must
```
```suggestion
To publish a list of active members on the website, the **Online Members Directory** module must
```

ksc-odoo replied:
please implement

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,91 @@
+Publish individual members
+--------------------------
+
+Return to :menuselection:`CRM app --> Sales --> Customers`, and click the Kanban card for a member.
```
```suggestion
Navigate to :menuselection:`CRM app --> Sales --> Customers`, and click the Kanban card for a member.
```

ksc-odoo reviewed (changes requested):
Hey @pabr-odoo -- just finished my Final Review of this PR. Great work. However, there is a section where I think some more information would help increase the value of this doc *and* help the reader/user understand the concept a little more thoroughly. So, if you could implement that adjustment, along with the other suggestions I left, and tag me again for another quick look, that would be great. Thanks!

pabr-odoo commented:
Hey @ksc-odoo thanks so much for your great review as always! I added the additional content to the members products section and implemented your changes. Please give it another look whenever you get the chance.

ksc-odoo reviewed (changes requested):
hey there @pabr-odoo -- just gave this another look. Noticed that most (if not all) of my previous comments were marked as 'Resolved,' but hadn't been implemented yet. I'm sure this was just a bug/mistake, so if you could ensure those modifications are made *before* moving this to the next stage, that would be great. Thanks!

pabr-odoo commented:
Oops so sorry about that @ksc-odoo, that was definitely an error on my part and not intentional. I made the changes locally but it seems I had an issue with the commit. It's looking correct on my end now, so please give it a look when you get the chance and let me know if anything else needs to be changed. Thanks so much!

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,108 @@
+===================
+
+To create a new membership product, navigate to :menuselection:`Members app --> Configuration -->
+Membership Products`, and click :guilabel:`New` to open a blank product record.
```
```suggestion
Membership Products`, and click :guilabel:`New` to open a blank product form.
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,108 @@
+
+On the new product form, enter a name for the product in the field under :guilabel:`Product Name`.
+Use the two blank fields to the right of :guilabel:`Membership Duration` to set a date from which
+membership becomes active. Finally, use the :guilabel:`Membership Fee` field to set the price at
```
```suggestion
the membership becomes active. Finally, use the :guilabel:`Membership Fee` field to set the price at
```

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,108 @@
+
+The following optional fields can also be configured for each product:
+
+An internal reference number can be set in the field to the right of :guilabel:`Internal Reference`,
+and a category set using the dropdown to the right of :guilabel:`Product Category`. Set a company
+using the field to the right of :guilabel:`Company` for the product to appear in that company's
+inventory. Use the slider to the right of :guilabel:`Active` to set hide a product without removing
+it.
+
+Tick the checkbox next to :guilabel:`Visible on current website` if desired, and use the field
+under :guilabel:`Description` for an internal product description, and :guilabel:`Sales Description`
+for a product description that will be visible on quotations. Use the field to the right of
+:guilabel:`Income Account` to use a different value than the default from the product category.
+Finally, set the default tax used when selling this product using the field next to
+:guilabel:`Taxes`.
```
While this info is correct, I feel like most docs with a section like this tend use a list format.

[name of field] : Explanation of the field.

That tends to be well received... so, consider that during your revisions.

The way it is currently written doesn't rely the information as cleanly (or as efficiently) as possible

ksc-odoo replied:
Upon further thought, I think you should *definitely* rework/reformat/reword this section so it is a cleaner, more digestible list. Doing so will add to the overall value of the doc. Thanks!

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,108 @@
+The following optional fields can also be configured for each product:
+
+An internal reference number can be set in the field to the right of :guilabel:`Internal Reference`,
+and a category set using the dropdown to the right of :guilabel:`Product Category`. Set a company
```
```suggestion
and a category set using the drop-down menu to the right of :guilabel:`Product Category`. Set a company
```
Drop-down is always hyphenated and followed by either "menu" or "window" -- depending on the circumstance

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,108 @@
+An internal reference number can be set in the field to the right of :guilabel:`Internal Reference`,
+and a category set using the dropdown to the right of :guilabel:`Product Category`. Set a company
+using the field to the right of :guilabel:`Company` for the product to appear in that company's
+inventory. Use the slider to the right of :guilabel:`Active` to set hide a product without removing
+it.
```
please rework/reword this portion about 'Active'

ksc-odoo commented on content/applications/sales/members.rst:
```diff
@@ -0,0 +1,108 @@
+
+.. tip::
+   To successfully buy a membership, both the :guilabel:`Account Receivable` and :guilabel:`Account
+   Payable` lines must first be configured for that contact. These can be found in the contact
```
```suggestion
   Payable` lines **must** be configured for that contact. These can be found in the contact
```

ksc-odoo reviewed (approved):
hey @pabr-odoo -- thanks for the quick fix. There are, however, a few more necessary adjustments that you'll need to address. So, once you make those changes, tag me again, and we should be able to move this to the next stage after another quick look. Thanks!

pabr-odoo commented:
Hey @ksc-odoo thanks for being so detailed, I completely agree about using a list format for the fields like you suggested. Please give it another look when you get the chance. Thanks!

ksc-odoo reviewed (approved):
Alrighty, @pabr-odoo -- just gave this another look. Seems good to me. Feel free to pass it on to the next stage, **after** making these last few modifications/adjustments. Thanks! 👍

pabr-odoo commented:
Hey @samueljlieber have another ready for tech review whenever you can get to it. Thanks!

samueljlieber reviewed (approved):
@pabr-odoo nice work on this backport, your PR looks good to me!
..
@robodoo r+ fw=no

robodoo commented:
Disabled forward-porting.
